### PR TITLE
Add new security audit reports

### PR DESCRIPTION
**Pull request by anegg0**
-   Add the latest security reports
-   Convert into a table for clarity + presentation
-   Host PDF versions since GitHub doesn't seem to render PDF reliably

### CODE REVIEW

File: arbitrum-docs/audit-reports.mdx [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+| `Trail of Bits`       | 24/07/2024 | `Arbos30 Nitro upgrade`          | [view](hosted-audit-reports/2024-04-offchain-arbos-30-nitro-upgrade-securityreview.pdf) | [source](https://github.com/trailofbits/publications/blob/master/reviews/2024-04-offchain-arbos-30-nitro-upgrade-securityreview.pdf) |
+| `Trail of Bits`       | 24/07/2024 | `Arbos 31`                       | [view](hosted-audit-reports/2024-04-offchain-arbos-31-securityreview.pdf)               | [source](https://github.com/trailofbits/publications/blob/master/reviews/2024-04-offchain-arbos-31-securityreview.pdf)               |
+| `Trail of Bits`       | 24/07/2024 | `l1-l3-teleporter`               | [view](hosted-audit-reports/2024-04-offchain-l1-l3-teleporter-securityreview.pdf)       | [source](https://github.com/trailofbits/publications/blob/master/reviews/2024-04-offchain-l1-l3-teleporter-securityreview.pdf)       |
+| `chainsecurity`       | -          | `l1-l3-teleporter`               | [view](https://www.chainsecurity.com/security-audit/offchain-labs-fund-distribution)    | [source](https://www.chainsecurity.com/security-audit/offchain-labs-fund-distribution)                                               |
```
Comment (leeederek) [inline arbitrum-docs/audit-reports.mdx:15]: the date here is march 20, 2024

File: arbitrum-docs/audit-reports.mdx [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-
-- [ ConsenSys Diligence, 11/5/2021: Core Contracts, Token Bridge](https://github.com/OffchainLabs/nitro/blob/master/audits/ConsenSys_Diligence_Arbitrum_Contracts_11_2021.pdf)
+| Auditor               | Audit date | Audited code                     | View report                                                                             | Source                                                                                                                               |
+|-----------------------|------------|----------------------------------|-----------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------|
+| `Code4rena`           | 6/17/2024  | `Arbitrum BoLD`                  | [view](https://code4rena.com/reports/2024-05-arbitrum-foundation)                       | [source](https://code4rena.com/reports/2024-05-arbitrum-foundation)                                                                  |
```
Comment (leeederek) [inline arbitrum-docs/audit-reports.mdx:5]:
```suggestion
| `Code4rena`           | 6/17/2024  | `Arbitrum BoLD: Public Audit Competition Report`                  | [view](https://code4rena.com/reports/2024-05-arbitrum-foundation)                       | [source](https://code4rena.com/reports/2024-05-arbitrum-foundation)                                                                  |
```

**Review by leeederek — APPROVED** [review]

File: arbitrum-docs/audit-reports.mdx (hunk 1 above)
Comment (leeederek) [inline arbitrum-docs/audit-reports.mdx:15]:
```suggestion
| `chainsecurity`       | -          | `Nova Fee Router Updates (ArbOS 31)`              | [view](https://www.chainsecurity.com/security-audit/offchain-labs-fund-distribution)    | [source](https://www.chainsecurity.com/security-audit/offchain-labs-fund-distribution)                                               |
```

Comment (anegg0) [reply arbitrum-docs/audit-reports.mdx]: Fixed, thank you!

Comment (anegg0) [reply arbitrum-docs/audit-reports.mdx]: fixed, tysm!

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**